purchases/src/main/kotlin/com/revenuecat/purchases/errors.kt

```diff
@@ -85,8 +85,12 @@ internal fun BackendErrorCode.toPurchasesError(underlyingErrorMessage: String) =
     PurchasesError(this.toPurchasesErrorCode(), underlyingErrorMessage)
 
 internal fun HTTPClient.Result.toPurchasesError(): PurchasesError {
-    val errorCode = body?.get("code") as? Int
-    val errorMessage = (body?.get("message") as? String) ?: ""
+    var errorCode: Int? = null
+    var errorMessage = ""
+    body?.let { body ->
+        errorCode = if (body.has("code")) body.get("code") as Int else null
+        errorMessage = if (body.has("message")) body.get("message") as String else ""
+    }
```

Comment:
I added this because when testing I realized that a 400 from the backend was giving a Network exception instead. The reason was that `body?.get("code")` was giving `JSONException`, and `JSONException`s give network errors.

Reply:
what happens when the backend is unreachable? i.e.: the device is disconnected

Reply:
That throws an IOException that goes to https://github.com/RevenueCat/purchases-android/pull/118/files/71aeb5f453b8bd0cafb5921a7eaf991bdcb4dbd3#diff-db1316d3a1f0983fcfa02b13653f47f3R75